src/staticanalysis/bot/static_analysis_bot/coverity/coverity.py

```diff
@@ -0,0 +1,257 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import json
+import os
+import subprocess
+
+from cli_common.log import get_logger
+from static_analysis_bot import COVERITY
+from static_analysis_bot import AnalysisException
+from static_analysis_bot import DefaultAnalyzer
+from static_analysis_bot import Issue
+from static_analysis_bot import stats
+from static_analysis_bot.config import settings
+from static_analysis_bot.revisions import Revision
+
+logger = get_logger(__name__)
+
+ISSUE_MARKDOWN = '''
+## coverity error
+
+- **Message**: {message}
+- **Location**: {location}
+- **In patch**: {in_patch}
+- **Coverity check**: {check}
+- **Publishable **: {publishable}
+- **Is new**: {is_new}
+
+```
+{body}
+```
+'''
+
+
+class Coverity(DefaultAnalyzer):
+    '''
+    Coverity runner
+    '''
+    def __init__(self, validate_checks=True):
+        # Ensure that we have all what we need into settings
+        assert settings.cov_package_name, 'Missing secret COVERITY_CONFIG:package_name'
+        assert settings.cov_package_ver, 'Missing secret COVERITY_CONFIG:package_ver'
+
+        self.cov_state_path = os.path.join(os.environ['MOZBUILD_STATE_PATH'],
+                                           'coverity')
+        self.cov_path = os.path.join(self.cov_state_path,
+                                     settings.cov_package_name)
+        self.cov_run_desktop = os.path.join(self.cov_path, 'bin', 'cov-run-desktop')
+        self.cov_translate = os.path.join(self.cov_path, 'bin', 'cov-translate')
+        self.cov_configure = os.path.join(self.cov_path, 'bin', 'cov-configure')
+        self.cov_work_path = os.path.join(self.cov_state_path, 'data-coverity')
+        self.cov_idir_path = os.path.join(self.cov_work_path, settings.cov_package_ver, 'idir')
+
+        assert os.path.exists(self.cov_path), \
+            'Missing Coverity in {}'.format(self.cov_path)
+
+    def get_files_with_commands(self):
+        '''
+        Returns an array of dictionaries having file_path with build command
+        '''
+
+        compile_db = json.loads(open(self.compile_commands_path, 'r').read())
```

Comment:
It's better to do `json.load(open(self.compile_commands_path))`

Note the missing **s** in `load`